### PR TITLE
feat: stylus support

### PR DESCRIPTION
**Pull request by alexkpek**
## I'm submitting a...

- [ ] Bug Fix
- [x] Feature
- [ ] Other (Refactoring, Added tests, Documentation, ...)

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern, see current [`.vcmrc`](https://github.com/dherges/ng-packagr/blob/master/.vcmrc)
- [x] Tests for the changes have been added

## Description

Added support for stylus preprocessor.
Tests provided.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/lib/steps/assets.ts [hunk 1]
```diff
@@ -36,7 +37,7 @@ export const processAssets = (src: string, dest: string): Promise<any> => {
         styleProcessor: (path, ext, file, cb) => {
 
           debug(`render stylesheet ${path}`);
-          const render = pickRenderer(path, ext, file);
+          const render = pickRenderer(path, ext, file, src);
```
Comment (dherges) [inline src/lib/steps/assets.ts]: Thanks for the pull request! Nice to have this feature.

One question: is the `src` location here used as the `ctx` for stylus to resolve relative imports?

I think it's a good idea to have a test for a relative import or at least put a code comment somewhere in this file.

Comment (alexkpek) [reply src/lib/steps/assets.ts]: Thanks for review.
I'll add a comment and test suite right now.

**Comment by alexkpek** [comment]
Update PR
Added more paths for stylus resolver (node_modules, current file path)
Added tests for resolving relative urls

**Comment by dherges** [comment]
Looks good!

I squash the change so that it's easier to track back in Git history the change introducing the stylus feature.

**Pull request merged**